pybamm/solvers/c_solvers/idaklu/IDAKLUSolverOpenMP.inl

```diff
@@ -295,21 +372,30 @@ Solution IDAKLUSolverOpenMP<ExprSet>::solve(
     }
   }
 
-  for (int i = 0; i < number_of_states; i++)
-  {
+  for (int i = 0; i < number_of_states; i++) {
     yval[i] = y0[i];
     ypval[i] = yp0[i];
   }
 
-  IDAReInit(ida_mem, t0, yy, yp);
-  if (number_of_parameters > 0)
-    IDASensReInit(ida_mem, IDA_SIMULTANEOUS, yyS, ypS);
+  SetSolverOptions();
+
+  CheckErrors(IDAReInit(ida_mem, t0, yy, yp));
+  if (sensitivity) {
+    CheckErrors(IDASensReInit(ida_mem, IDA_SIMULTANEOUS, yyS, ypS));
+  }
 
   // correct initial values
-  DEBUG("IDACalcIC");
-  IDACalcIC(ida_mem, IDA_YA_YDP_INIT, t(1));
-  if (number_of_parameters > 0)
-    IDAGetSens(ida_mem, &t0, yyS);
+  int const init_type = solver_opts.init_all_y_ic ? IDA_Y_INIT : IDA_YA_YDP_INIT;
+  if (solver_opts.calc_ic) {
+    DEBUG("IDACalcIC");
+    // Do not throw a warning if the initial conditions calculation fails
```

Comment:
I think if the user has requested the initial condition be calculated, they would probably want to know if it fails (just a warning is fine, doesn't have to abort)

Reply:
I will reword this comment. IDA already shows messages if its functions fail, so I meant that I did not want to show a duplicate warning message here. For example, running [this IDAKLU unit test](https://github.com/MarcBerliner/PyBaMM/blob/89a48b02b2d21ddd407743b34ea65a69390b4d52/tests/unit/test_solvers/test_idaklu_solver.py#L471) returns the following warnings and errors
```python
[IDAS ERROR]  IDACalcIC
  The linesearch algorithm failed: step too small or too many backtracks.


[IDAS ERROR]  IDASolve
  At t = 0 and h = 1.14441e-08, the corrector convergence failed repeatedly or with |h| = hmin.

Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/marcberliner/Documents/GitHub/PyBaMM/pybamm/solvers/base_solver.py", line 901, in solve
    new_solution = self._integrate(
                   ^^^^^^^^^^^^^^^^
  File "/Users/marcberliner/Documents/GitHub/PyBaMM/pybamm/solvers/idaklu_solver.py", line 945, in _integrate
    raise pybamm.SolverError("idaklu solver failed")
pybamm.expression_tree.exceptions.SolverError: idaklu solver failed
```